### PR TITLE
Karim's fast-sdxl changes

### PR DESCRIPTION
**Pull request by emrebayramc**
Karim's changes from yesterday

Add refiner model for last 20% of steps

Revert AutoPipeline changes

Karim's changes from yesterday

Add refiner model for last 20% of steps

Revert AutoPipeline changes

Allow arbitrary types

Always send images even if we will drop

Cleanup

Add more logging & error detection

Fix miner import and init errors

Switch to non deprecated batches URL

Fix str upload

Fix upload masking

Rever output_type to fix nsfw error

Fix refiner errors

All optional refiner

### CODE REVIEW

File: neurons/miners/StableMiner/stable_miner.py [hunk 1]
```diff
@@ -1,93 +1,112 @@
...
-            logger.error(e)
-            raise
+        return self.miner_config.model_configs[model_type][task_type]
 
     def load_model(self, model_name: str, task_type: TaskType) -> torch.nn.Module:
         try:
             logger.info(f"Loading model {model_name} for task {task_type}...")
-            config = self.task_configs[task_type]
-            model_loader = ModelLoader(self.config.miner)
+            config = next(tc for tc in self.task_configs if tc.task_type == task_type)
```
Comment (tavurth) [inline neurons/miners/StableMiner/stable_miner.py]: Ternary loops can be pretty hard to read, maybe let's do:

```py
def get_config_for_task_type(self, task_type):
    for config in self.task_configs:
        if config.task_type == task_type:
            return config
    
    # If no matching config is found, raise an exception
    raise ValueError(f"No configuration found for task type: {task_type}")
```

& then try-except

**Review by tavurth — APPROVED** [review]
